irohad/ametsuchi/impl/postgres_command_executor.cpp

```diff
@@ -166,21 +169,38 @@ namespace {
     return makeCommandError(std::move(command_name), 1, std::move(query_args));
   }
 
+  template <typename T>
+  std::string permissionSetToBitString(
+      const shared_model::interface::PermissionSet<T> &set) {
+    return (boost::format("'%s'") % set.toBitstring()).str();
+  }
+
   std::string checkAccountRolePermission(
-      Role permission,
+      const std::string &permission_bitstring,
       const shared_model::interface::types::AccountIdType &account_id) {
-    const auto perm_str =
-        shared_model::interface::RolePermissionSet({permission}).toBitstring();
     std::string query = (boost::format(R"(
-          SELECT COALESCE(bit_or(rp.permission), '0'::bit(%1%))
-          & '%2%' = '%2%' FROM role_has_permissions AS rp
+          SELECT
+              COALESCE(bit_or(rp.permission), '0'::bit(%1%))
+              & (%2%::bit(%1%) | '%3%'::bit(%1%))
```

Comment:
Why do you need those `::bit(%1)` ?
It worked well without them.

Reply:
I think it's a part of my proposed patch. I just prefer explicit things when they cost (almost) nothing. But I don't mind changing it to auto type if it works. Also I missed single quotes around `%2%`.

Reply:
We need one of `::bit(%1)`
Without both, error occurs
`HINT:  Could not choose a best candidate operator. You might need to add explicit type casts.`

About `%2%` without single quotes. the argument is bitstring that cast into a string, so it is normal